### PR TITLE
Concrete walls

### PR DESCRIPTION
**Pull request by CypressBz**
## About The Pull Request

Adding a concrete wall that is neither deconstruct-able, or build-able, but is not impervious to damage. Strictly for ruin mapping for now.

image

## Why It's Good For The Game

ruin map flow can be gamed hard with a toolbox full of tools. this prevents that. plus it looks really nice.

## Changelog
:cl:
add: Adds some new concrete walls!
/:cl:

### CODE REVIEW

**Comment by CypressBz** [comment]
oh cool, I'll figure it out tomorrow.

File: code/game/turfs/closed/wall/misc_walls.dm [hunk 1]
```diff
@@ -110,3 +110,24 @@
...
+	base_icon_state = "concrete_wall"
+	girder_type = /obj/structure/girder/reinforced
+	hardness = 10
+	explosion_block = 2
+	rad_insulation = RAD_HEAVY_INSULATION
+
+/turf/closed/wall/concrete/deconstruction_hints(mob/user)
+	switch(d_state)
+		if(INTACT)
+			return "<span class='notice'>Nothing's going to cut that.</span>"
```
Comment (atakiya) [inline code/game/turfs/closed/wall/misc_walls.dm:128]:
```suggestion
/turf/closed/wall/concrete/deconstruction_hints(mob/user)
	return "<span class='notice'>Nothing's going to cut that.</span>"
```
No need for the checks here if this is indestructible.

**Comment by atakiya** [comment]
Please undraft once this is ready to be reviewed again.

File: code/game/turfs/closed/wall/misc_walls.dm (hunk 1 above)
Comment (CypressBz) [reply code/game/turfs/closed/wall/misc_walls.dm:128]: it's not indestructable, that's the whole point. but also I don't think it has any states other than intact.

Comment (atakiya) [reply code/game/turfs/closed/wall/misc_walls.dm:128]: > it's not indestructable, that's the whole point. but also I don't think it has any states other than intact.

This type of wall actually does not have any `d_state`s at all - I believe its short for "deconstruction_state".
Only the `r_wall` type does, which yours is not a subtype (child) of.

This isn't a problem though, as if this is the case, you can just skip this check and immediately show that message to the user.

**Comment by CypressBz** [comment]
Will fix in like an hour or two, I have no idea how to read linters' output.

**Review by atakiya — APPROVED** [review]
All good now.

**Comment by FedTheCat** [comment]
Good job! Love these.

**Pull request merged**